force-app/main/default/classes/FSL/WorkOrderImportModel.cls

```diff
@@ -0,0 +1,189 @@
+            get; set;
+        }
+        @AuraEnabled
+        public String ScheduledStartTime {
+            get; set;
+        }
+        @AuraEnabled
+        public String ScheduledFinishTime {
+            get; set;
+        }
+        public String validate() {
+            if (VendorId == null)
+                return 'Vendor ID';
+            if (EmployeeCode == null)
+                return 'Employee Code';
+            if (Status == null)
+                return 'Status';
+            if (ScheduledStartDate == null)
+                return 'Scheduled Start Date';
+            if (ScheduledFinishDate == null)
+                return 'Scheduled Finish Date';
+            if (ScheduledStartTime == null)
+                return 'Scheduled Start Time';
+            if (ScheduledFinishTime == null)
+                return 'Scheduled Finish Time';
+            return null;
+        }
+    }
+
+    @AuraEnabled
+    public List<WorkOrderExcel> WorkOrders {
+        get; set;
+    }
+    @AuraEnabled
+    public List<WorkOrderLineItemExcel> WorkOrderLineItems {
+        get; set;
+    }
+    @AuraEnabled
+    public List<ServiceAppointmentExcel> ServiceAppointments {
+        get; set;
+    }
+    public String validate() {
+        if (WorkOrders == null || WorkOrders.size() < 1) {
+            return 'No Work Orders data was found in spreadsheet.';
+        }
+        if (WorkOrderLineItems == null || WorkOrderLineItems.size() < 1) {
+            return 'No Work Order Line Items data was found in spreadsheet.';
+        }
+        if (ServiceAppointments == null || ServiceAppointments.size() < 1) {
+            return 'No Service Appointments data was found in spreadsheet.';
+        }
+        String message;
+        for (WorkOrderExcel w : WorkOrders) {
+            message = w.validate();
+            if (message != null) {
+                return String.format(
+                    'At least one data is missing, required field: "{0}", ' +
+                    'in line {1}',
+                    new List<String> { message, w.FakeId }
+                    );
+            }
+        }
+        for (WorkOrderLineItemExcel w : WorkOrderLineItems) {
+            message = w.validate();
+            if (message != null) {
+                return String.format(
+                    'At least one Work Order Line Item is missing, ' +
+                    'required field: "{0}", in line {1}',
+                    new List<String> { message, w.FakeId }
+                    );
+            }
+        }
+        for (ServiceAppointmentExcel s : ServiceAppointments) {
+            message = s.validate();
+            if (message != null) {
+                return String.format(
+                    'At least one Service Appointment is missing, ' +
+                    'required field: "{0}", in line {1}',
+                    new List<String> { message, s.FakeId }
+                    );
+            }
+        }
```

Comment:
WorkOrderExcel, WorkOrderLineItemExcel, ServiceAppointmentExcel should have common super class (or interface, but here i'd prefer to have super class) that would gather common properties and have abstract validate method - to be implemented for each class separately.
With this approach we could have only one loop over only one collection - string error message can be embedded into the classes.